Dockerfile

```diff
@@ -0,0 +1,9 @@
+FROM python:3.10
+RUN curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/install-poetry.py | python -
+ENV PATH=$PATH:/root/.local/share/pypoetry/venv/bin/
+COPY pyproject.toml poetry.toml /app/
+WORKDIR /app
+RUN poetry install
+COPY .env.template /app/
+COPY todo_app /app/todo_app
+ENTRYPOINT poetry run gunicorn --bind 0.0.0.0 "todo_app.app:create_app()"
```

Comment:
Your dockerfile contains a single stage. You should have at least 2: one for dev (running flask) and one for prod (running gunicorn).
You can take advantage of docker compose to only build a certain target for a particular service. At the moment, your docker compose is only building a production service running gunicorn